### ISSUE TITLE
How to setup on VSCode

### ISSUE THREAD

**Issue by MohammedUbaid-eaton** [bneumann/CppUTest-Test-Adapter#26]
Hi,

I installed CppUTest on vscode & also installed dependency Test UI Explorer

![image](https://user-images.githubusercontent.com/96242355/147203665-7ae012c8-1b67-4421-834b-04a2221a15bc.png)

I do not understand how to get started.

I have a .cpp file on which i need to run unit tests,

Kindly guide.

**Comment by bneumann** [comment]
Hi, 
you first need to set up the paths in the settings as described here: https://github.com/bneumann/CppUTest-Test-Adapter#setup

After you set the test executable path the plugin will automatically detect the tests inside it and show the in the test explorer

**Comment by steverb1** [comment]
I'm not clear what is meant by "test executable".  What kind of file would that be?  (.exe or something else?). Is this a file CppUTest provides?

**Comment by bneumann** [comment]
Well if you compile your test code you should get an executable (i guess the one you are running?). That is your test executable. If you set it in the paths of this extension it should be used to identify your tests.

**Comment by steverb1** [comment]
OK, thanks, I found the executable, and have settings.json correctly configured now.  Still no tests showing up in the Test Explorer.  Any ideas about what else might be wrong?

**Comment by bneumann** [comment]
You are completely right, I just tried it again (after years to be honest) and it didn't work. I will look into it

**Comment by bneumann** [comment]
Okay I found the culprit and will release a new version today evening. Funny enough I think not a lot people are using this extension because you could actually add multiple files likes so
`"cpputestTestAdapter.testExecutable": "test1.exe;test2.exe"`
but not like this
`"cpputestTestAdapter.testExecutable": "test1.exe"`
I never wrote tests (shame on me) for the settings parser, so one of the last PRs got through breaking the default behaviour.
Thanks for adressing this issue

**Comment by steverb1** [comment]
Thank you very much for finding the problem!  Looking forward to the fix in the new version.

**Comment by bneumann** [comment]
No problem, i am a bit short on time recently (like in the past 2 years I guess). So it always takes some time to get things done. I just uploaded the latest version 1.3.1. Let me know if it works now

**Comment by steverb1** [comment]
OK, that is definitely an improvement.  I'm seeing things in the test explorer for the first time.  However, I'm now getting an ERROR_LOADING_TESTS

Here is what I have in settings.json:
"cpputestTestAdapter.testExecutable": "${workspaceFolder}/t0/t0_tests",
"cpputestTestAdapter.testExecutablePath": "${workspaceFolder}/t0"

**Comment by steverb1** [comment]
![Screenshot 2024-04-17 at 09 50 19](https://github.com/bneumann/CppUTest-Test-Adapter/assets/20443323/2b32f237-e1f2-4f86-807b-eb429508ad47)

**Comment by steverb1** [comment]
That double quote character at the end looks suspicious.  Would that still be something going on with the parser?

**Comment by bneumann** [comment]
That is odd, this text is loaded from the executable name itself. it worked with my local test executable file. Is the log showing anything helpful? I will check the commits if i fat fingered something in the code :)

**Comment by steverb1** [comment]
Any update on this?  And sorry, I just noticed you aksed if the log is showing anything.  Where would I find that?

**Comment by msoleck** [comment]
I have similar problems with this issue.

Here is the software I am running: 
- OS: Windows 11
- VS Code: 1.90.2
- CppUTest Test Adapter: 1.3.4

I previously had the **ERROR LOADING TESTS** error, but I resolved that by pointing to the correct executable.

I now get this other weird message:
![image](https://github.com/bneumann/CppUTest-Test-Adapter/assets/96546634/85d36a62-7403-4e05-abbd-98d1797da041)

I get a similar "error" message when i run it from the terminal
![image](https://github.com/bneumann/CppUTest-Test-Adapter/assets/96546634/7c1d1cb0-2b74-4e0c-8060-2b230e4fc316)

Here is the result in the output window
![image](https://github.com/bneumann/CppUTest-Test-Adapter/assets/96546634/0d1ae87d-52c2-4602-9fad-d28a07bb51be)

When I run the executable without the parameters, here is the output:
![image](https://github.com/bneumann/CppUTest-Test-Adapter/assets/96546634/ae8070d1-af3f-4f62-b5c9-34f4f68491d6)

My belief is that the tool tries to run the executable with some parameters.

My question is: does the this tool add some parameters which I am not aware?

Thank you for any help you can provide

**Comment by bneumann** [comment]
Hi, so the extension tries to get the list of tests from your executable. This is done by calling either `testrunner.exe -ll` or `testrunner.exe -ln` 
Which version of CppuTest are you using?

**Comment by steverb1** [comment]
4.0

**Comment by bneumann** [comment]
Weird, that is the same as mine. Can you add the logs? This can be done by setting the `cpputestTestAdapter.logpanel` propery to true.

**Comment by steverb1** [comment]
OK, once I set logpanel to true, where would I find the log file?
Also, I'm not sure there would be any logs, as no tests are being run.  (That is my original problem: cpputestTestAdapter isn't finding any tests).

**Comment by bneumann** [comment]
Sorry, yeah you will find it as described here:
https://github.com/bneumann/CppUTest-Test-Adapter/issues/65#issuecomment-2395409515
